src/index.ts

```diff
@@ -1,2 +1,143 @@
-const msg: string = "hi";
-console.log(msg);
\ No newline at end of file
+enum NoteType {
+    DEFAULT = "Default",
+    CONFIRMATION_REQUIRED = "Confirmation Required"
+}
+
+class TodoItem {
+    private static currentId = 0;
+    public readonly id: number;
+    public title: string;
+    public description: string;
+    public creationDate: Date;
+    public lastEditDate: Date;
+    public isCompleted: boolean;
+    public isEditable: boolean;
+    public type: NoteType;
+
+    constructor(title: string, description: string, type: NoteType, isEditable: boolean = false) {
+        if (!title.trim() || !description.trim()) {
+            throw new Error("Title and description cannot be empty.");
+        }
+        this.id = TodoItem.currentId++;
+        this.title = title;
+        this.description = description;
+        this.creationDate = new Date();
+        this.lastEditDate = new Date();
+        this.isCompleted = false;
+        this.isEditable = isEditable;
+        this.type = type;
+    }
+
+    // Метод для позначення нотатки як виконаної
+    markAsCompleted(): void {
+        this.isCompleted = true;
+    }
+
+    // Метод для оновлення дати останнього редагування
+    updateLastEditDate(): void {
+        this.lastEditDate = new Date();
+    }
+}
+
+class TodoList {
+    private items: TodoItem[] = [];
+
+    addItem(title: string, description: string, type: NoteType = NoteType.DEFAULT): TodoItem {
+        const newItem = new TodoItem(title, description, type);
+        this.items.push(newItem);
+        return newItem;
+    }
+
+    // Метод для видалення запису за ідентифікатором
+    removeItem(id: number): boolean {
+        const index = this.items.findIndex(item => item.id === id);
+        if (index !== -1) {
+            this.items.splice(index, 1);
+            return true;
+        }
+        return false;
+    }
+
+    // Метод для редагування запису за ідентифікатором
+    editItem(id: number, title: string, description: string, type: NoteType): boolean {
```

Comment:
It would be better to implement a return function value as the value of the new or previous item. Boolean type is not quite informative.